fdbserver/ClusterController.actor.cpp

```diff
@@ -208,11 +209,13 @@ ACTOR Future<Void> handleLeaderReplacement(Reference<ClusterRecoveryData> self,
 ACTOR Future<Void> clusterWatchDatabase(ClusterControllerData* cluster,
                                         ClusterControllerData::DBInfo* db,
                                         ServerCoordinators coordinators,
-                                        Future<Void> leaderFail) {
+                                        Future<Void> leaderFail,
+                                        Future<Void> recoveredDiskFiles) {
 	state MasterInterface iMaster;
 	state Reference<ClusterRecoveryData> recoveryData;
 	state PromiseStream<Future<Void>> addActor;
 	state Future<Void> recoveryCore;
+	state bool recoveredDisk = false;
```

Comment:
I think futures can be waited on multiple times? In that case I think we don't need to check if it is the first time we are waiting for `recoveredDiskFiles`.

Reply:
I tried to keep the same semantics as the existing code, in existing code it can be waited only once, so kept it same. Would like to keep it this way such that "non-Encryption" and "Encryption" execution semantics remains same (incase any future changes impacts the behavior)
Thanks

Reply:
okay for the `recoveredDisk` boolean check. But I'm against having encryption and non-encryption behave differently. It makes the code harder to maintain and less well tested. We should just move the `recoveredDiskFiles` wait here for all cases.

Reply:
Well in that case it would be better not to even do `Encryption` check, or in other words, we change the code execution semantics for `recoveredDiskFiles`. 

Also, as mentioned in my comment below, the intention is not avoid updating `non-Encryption` workflow unless Encryption codepath is sufficiently tested; I will add a TODO to make it more readable, but, my proposal is to simplify the code after getting some soak on Encryption codepath and make it one (no Encryption checks). Approach also minimizes impact crater (non Encryption continues to behave same) incase more work is needed with this patch.

Does it sounds reasonable?